lib/ssh_scan/scan_engine.rb

```diff
@@ -71,13 +71,26 @@ def scan_target(socket, opts)
           raise e
         end
       else
+        result['fingerprints'] = {}
         pkey = SSHScan::Crypto::PublicKey.new(host_key)
-        if pkey.is_supported?
-          result['fingerprints'] = {
-            "md5" => pkey.fingerprint_md5,
-            "sha1" => pkey.fingerprint_sha1,
-            "sha256" => pkey.fingerprint_sha256,
-          }
+        if pkey.is_rsa_supported?
+          result['fingerprints'].merge!({
+            "rsa" => {
+              "md5" => pkey.fingerprint_md5,
+              "sha1" => pkey.fingerprint_sha1,
+              "sha256" => pkey.fingerprint_sha256,
+            }
+          })
+        end
+
+        if pkey.is_dsa_supported?
+          result['fingerprints'].merge!({
+            "dsa" => {
+              "md5" => pkey.fingerprint_md5,
+              "sha1" => pkey.fingerprint_sha1,
+              "sha256" => pkey.fingerprint_sha256,
+            }
+          })
```

Comment:
I'm beginning to wonder if we need to make multiple connections to determine RSA and DSA keys on the same server, so I asked for example that shows both.  I feel like that will tease out whether that's needed or not.